Sources/Commands/SwiftTool.swift

```diff
@@ -485,27 +485,21 @@ public class SwiftTool {
         return try self.getNetrcConfig()?.get()
     }
 
-    func getNetrcConfig() throws -> Workspace.Configuration.Netrc? {
-        guard options.netrc || options.netrcFilePath != nil || options.netrcOptional else {
-            return .none
-        }
+    func getNetrcConfig() -> Workspace.Configuration.Netrc? {
+        guard options.netrc else { return nil }
 
-        let netrcFilePath = try self.netrcFilePath()
-        return netrcFilePath.map { .init(path: $0, fileSystem: localFileSystem) }
-    }
-
-    private func netrcFilePath() throws -> AbsolutePath? {
-        let netrcFilePath = options.netrcFilePath ?? localFileSystem.homeDirectory.appending(component: ".netrc")
-        guard localFileSystem.exists(netrcFilePath) else {
-            if !options.netrcOptional {
-                ObservabilitySystem.topScope.emit(error: "Cannot find mandatory .netrc file at \(netrcFilePath). To make .netrc file optional, use --netrc-optional flag.")
-                throw ExitCode.failure
-            } else {
-                ObservabilitySystem.topScope.emit(warning: "Did not find optional .netrc file at \(netrcFilePath).")
-                return .none
+        if let configuredPath = options.netrcFilePath {
+            guard localFileSystem.exists(configuredPath) else {
+                ObservabilitySystem.topScope.emit(warning: "Did not find .netrc file at \(configuredPath).")
+                return nil
             }
+
+            return .init(path: configuredPath, fileSystem: localFileSystem)
+        } else {
+            let defaultPath = localFileSystem.homeDirectory.appending(component: ".netrc")
+
+            return .init(path: defaultPath, fileSystem: localFileSystem)
```

Comment:
I think the current downstream code expectation is that this func will return nil if the file is not found, just like above, just without the warning

```
guard localFileSystem.exists(defaultPath) else {
  return .none
}
```

Reply:
Thanks for decoding that error for me. This is fixed by 4573b0ac0e269b3c6b153a60ccdf9552f16c7c09.